client/main.go

```diff
@@ -0,0 +1,1854 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"sync"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/gookit/color"
+)
+
+const (
+	Version          = "3.12.0.240221" // Overwritten automatically by -ldflags
+	ReportTimeout    = 3 * time.Minute
+	OAUTH_CLIENT_ID  = "IdFRwa3SGA8eMpzhRVFMg5Ts8sPK93xBjif93x0F"
+	WindowsPathLimit = 250
+
+	// PATHS
+	server_default     = "https://www.blenderkit.com" // default address to production blenderkit server
+	gravatar_dirname   = "bkit_g"                     // directory in safeTempDir() for gravatar images
+	cleanfile_path     = "blendfiles/cleaned.blend"   // relative path to clean blend file in add-on directory
+	upload_script_path = "upload_bg.py"               // relative path to upload script in add-on directory
+
+	// EMOJIS
+	EmoOK            = "✅"
+	EmoCancel        = "⛔"
+	EmoWarning       = "⚠️ " // Needs space at the end for proper alignment, not sure why.
+	EmoInfo          = "ℹ️ "
+	EmoError         = "❌"
+	EmoNetwork       = "📡"
+	EmoNewConnection = "🤝"
+	EmoDisconnecting = "👐"
+	EmoSecure        = "🔒"
+	EmoInsecure      = "🧨"
+)
+
+var (
+	SystemID        *string // Unique ID of the current system (15 integers)
+	PlatformVersion string
+	Port            *string
+	Server          *string
+
+	CodeVerifier    string //Used for OAuth2
+	CodeVerifierMux sync.Mutex
+
+	lastReportAccess    time.Time
+	lastReportAccessMux sync.Mutex
+
+	ActiveAppsMux sync.Mutex
+	ActiveApps    []int
+
+	Tasks                map[int]map[string]*Task
+	TasksMux             sync.Mutex
+	AddTaskCh            chan *Task
+	TaskProgressUpdateCh chan *TaskProgressUpdate
+	TaskMessageCh        chan *TaskMessageUpdate
+	TaskFinishCh         chan *TaskFinish
+	TaskErrorCh          chan *TaskError
+	TaskCancelCh         chan *TaskCancel
+
+	ClientAPI, ClientDownloads, ClientUploads, ClientSmallThumbs, ClientBigThumbs *http.Client
+
+	BKLog   *log.Logger
+	ChanLog *log.Logger
+)
+
+func init() {
+	Tasks = make(map[int]map[string]*Task)
+	AddTaskCh = make(chan *Task, 100)
+	TaskProgressUpdateCh = make(chan *TaskProgressUpdate, 1000)
+	TaskMessageCh = make(chan *TaskMessageUpdate, 1000)
+	TaskFinishCh = make(chan *TaskFinish, 100)
+	TaskCancelCh = make(chan *TaskCancel, 100)
+	TaskErrorCh = make(chan *TaskError, 100)
+
+	PlatformVersion = runtime.GOOS + " " + runtime.GOARCH + " go" + runtime.Version()
```

Comment:
BTW, have you noticed the `go` string is duplicated in the header, like `linux amd64 gogo1.21.6` or `darwin arm64 gogo1.21.6`? I mean, I have no objection, just wanted to make sure you are aware of that.

Reply:
Extra go is a mistake, gonna fix it.